### ISSUE TITLE
[Bug]: multi cn e2e bvt runs timed out

### ISSUE THREAD

**Issue by yingfeng** [matrixorigin/matrixone#6421]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93):
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

Multi CN E2E bvt run timed out

First get auth timed out:

```
1, cmd: 0, data: , err: "}
2022/11/08 13:02:27.754952 +0000 INFO frontend/util.go:510 time of Exec.Run : 882.019µs connectionId 1001
2022/11/08 13:02:27.755013 +0000 INFO frontend/util.go:495 query trace status {"connection_id": 1001, "statement": "select task_id, task_metadata_id, task_metadata_executor
, task_metadata_context, task_metadata_option, task_parent_id, task_status, task_runner, task_epoch, last_heartbeat, result_code, error_msg, create_at, end_at from mo_task.
sys_async_task where task_status = 0 order by task_id", "status": "success", "session_info": "connectionId 1001"}
2022/11/08 13:02:29.385298 +0000 INFO compile/cnserviceDriver.go:227 error: rpc timeout {"span": {}}
2022/11/08 13:02:29.385398 +0000 ERROR frontend/util.go:505 rpc timeout {"session_info": ""}
github.com/matrixorigin/matrixone/pkg/frontend.logError
        /home/runner/work/matrixone/matrixone/head/pkg/frontend/util.go:505
github.com/matrixorigin/matrixone/pkg/frontend.(*MysqlCmdExecutor).doComQuery
        /home/runner/work/matrixone/matrixone/head/pkg/frontend/mysql_cmd_executor.go:3026
github.com/matrixorigin/matrixone/pkg/frontend.(*BackgroundHandler).Exec
        /home/runner/work/matrixone/matrixone/head/pkg/frontend/session.go:1863
github.com/matrixorigin/matrixone/pkg/frontend.determineRoleSetHasPrivilegeSet.func1
        /home/runner/work/matrixone/matrixone/head/pkg/frontend/authenticate.go:3469
github.com/matrixorigin/matrixone/pkg/frontend.determineRoleSetHasPrivilegeSet
        /home/runner/work/matrixone/matrixone/head/pkg/frontend/authenticate.go:3507
github.com/matrixorigin/matrixone/pkg/frontend.determineUserHasPrivilegeSet
        /home/runner/work/matrixone/matrixone/head/pkg/frontend/authenticate.go:3631
github.com/matrixorigin/matrixone/pkg/frontend.authenticateUserCanExecuteStatementWithObjectTypeAccountAndDatabase
        /home/runner/work/matrixone/matrixone/head/pkg/frontend/authenticate.go:3999
github.com/matrixorigin/matrixone/pkg/frontend.authenticateUserCanExecuteStatement
        /home/runner/work/matrixone/matrixone/head/pkg/frontend/mysql_cmd_executor.go:2312
github.com/matrixorigin/matrixone/pkg/frontend.(*MysqlCmdExecutor).doComQuery
        /home/runner/work/matrixone/matrixone/head/pkg/frontend/mysql_cmd_executor.go:2461
github.com/matrixorigin/matrixone/pkg/frontend.(*MysqlCmdExecutor).ExecRequest
        /home/runner/work/matrixone/matrixone/head/pkg/frontend/mysql_cmd_executor.go:3103
github.com/matrixorigin/matrixone/pkg/frontend.(*Routine).Loop
        /home/runner/work/matrixone/matrixone/head/pkg/frontend/routine.go:159
2022/11/08 13:02:29.385547 +0000 ERROR frontend/util.go:505 query trace status {"connection_id": 4294967295, "statement": "select rp.privilege_id, rp.with_grant_option from mo_catalog.mo_database as d cross join mo_catalog.mo_role_privs as rp where d.dat_id = rp.obj_id and rp.obj_type = \"database\" and rp.role_id = 0 and rp.privilege_id = 20 and rp.privilege_level = \"d\" and d.datname = \"func_string_space\"", "status": "fail", "error": "rpc timeout", "session_info": ""}
github.com/matrixorigin/matrixone/pkg/frontend.logError
        /home/runner/work/matrixone/matrixone/head/pkg/frontend/util.go:505
github.com/matrixorigin/matrixone/pkg/frontend.logStatementStringStatus
        /home/runner/work/matrixone/matrixone/head/pkg/frontend/util.go:489
```

Then a bags of get db error 
```
2008-12-04\"),(4, \"2012-03-23\");", "session_info": "connectionId 1005"}
2022/11/08 13:02:29.434298 +0000 INFO frontend/session.go:1514 error: not connect to a database {"span": {}}
2022/11/08 13:02:29.434325 +0000 INFO plan/build_insert.go:54 error: invalid input: insert table is invalid 't1' {"span": {}}
2022/11/08 13:02:29.434338 +0000 ERROR frontend/util.go:505 invalid input: insert table is invalid 't1' {"session_info": "connectionId 1005"}
github.com/matrixorigin/matrixone/pkg/frontend.logError
        /home/runner/work/matrixone/matrixone/head/pkg/frontend/util.go:505
github.com/matrixorigin/matrixone/pkg/frontend.(*MysqlCmdExecutor).doComQuery
        /home/runner/work/matrixone/matrixone/head/pkg/frontend/mysql_cmd_executor.go:3026
github.com/matrixorigin/matrixone/pkg/frontend.(*MysqlCmdExecutor).ExecRequest
        /home/runner/work/matrixone/matrixone/head/pkg/frontend/mysql_cmd_executor.go:3103
github.com/matrixorigin/matrixone/pkg/frontend.(*Routine).Loop
        /home/runner/work/matrixone/matrixone/head/pkg/frontend/routine.go:159
2022/11/08 13:02:29.434385 +0000 ERROR frontend/util.go:505 query trace status {"connection_id": 1005, "statement": "insert into t1 values (1, \"2012-10-12\"), (1, \"2004-04-24\"), (3, \"2008-12-04\"), (4, \"2012-03-23\")", "status": "fail", "error": "invalid input: insert table is invalid 't1'", "session_info": "connectionId 1005"}
github.com/matrixorigin/matrixone/pkg/frontend.logError
        /home/runner/work/matrixone/matrixone/head/pkg/frontend/util.go:505
github.com/matrixorigin/matrixone/pkg/frontend.logStatementStringStatus
        /home/runner/work/matrixone/matrixone/head/pkg/frontend/util.go:489
github.com/matrixorigin/matrixone/pkg/frontend.logStatementStatus
        /home/runner/work/matrixone/matrixone/head/pkg/frontend/util.go:481
github.com/matrixorigin/matrixone/pkg/frontend.(*MysqlCmdExecutor).doComQuery
        /home/runner/work/matrixone/matrixone/head/pkg/frontend/mysql_cmd_executor.go:3034
github.com/matrixorigin/matrixone/pkg/frontend.(*MysqlCmdExecutor).ExecRequest
        /home/runner/work/matrixone/matrixone/head/pkg/frontend/mysql_cmd_executor.go:3103
github.com/matrixorigin/matrixone/pkg/frontend.(*Routine).Loop
        /home/runner/work/matrixone/matrixone/head/pkg/frontend/routine.go:159
```

Overall logs: [multi-cn-e2e_linux_x86_reports.zip](https://github.com/matrixorigin/matrixone/files/9963160/multi-cn-e2e_linux_x86_reports.zip)

### Steps to Reproduce

```Markdown
https://github.com/matrixorigin/matrixone/actions/runs/3419586588/jobs/5693296611
```

**Comment by sukki37** [comment]
`get db error` is caused by not specifying the database when mo-tester re-establishes the connection, and Su-Dong @aressu1985  will fix this bug soon.

**Comment by sukki37** [comment]
Update: lots of `rpc timeout` and `illegal state  connectionId` exists.
```xshell
2022/11/08 13:48:32.302214 +0000 INFO frontend/util.go:495 query trace status {"connection_id": 1002, "statement": "select task_id, task_metadata_id, task_metadata_executor, task_metadata_context, task_metadata_option, task_parent_id, task_status, task_runner, task_epoch, last_heartbeat, result_code, error_msg, create_at, end_at from mo_task.sys_async_task where task_id > 1 and task_runner = \"dd1dccb5-4d3c-41f8-b482-5251dc7a41bf\" order by task_id limit 1", "status": "success", "session_info": "connectionId 1002"}
2022/11/08 13:48:32.338017 +0000 INFO compile/cnserviceDriver.go:227 error: rpc timeout {"span": {}}
2022/11/08 13:48:32.338100 +0000 ERROR frontend/util.go:505 rpc timeout {"session_info": ""}
github.com/matrixorigin/matrixone/pkg/frontend.logError
	/home/runner/work/matrixone/matrixone/head/pkg/frontend/util.go:505
github.com/matrixorigin/matrixone/pkg/frontend.(*MysqlCmdExecutor).doComQuery
	/home/runner/work/matrixone/matrixone/head/pkg/frontend/mysql_cmd_executor.go:3026
github.com/matrixorigin/matrixone/pkg/frontend.(*BackgroundHandler).Exec
	/home/runner/work/matrixone/matrixone/head/pkg/frontend/session.go:1863
github.com/matrixorigin/matrixone/pkg/frontend.determineRoleSetHasPrivilegeSet.func1
	/home/runner/work/matrixone/matrixone/head/pkg/frontend/authenticate.go:3469
github.com/matrixorigin/matrixone/pkg/frontend.determineRoleSetHasPrivilegeSet
	/home/runner/work/matrixone/matrixone/head/pkg/frontend/authenticate.go:3507
github.com/matrixorigin/matrixone/pkg/frontend.determineUserHasPrivilegeSet
	/home/runner/work/matrixone/matrixone/head/pkg/frontend/authenticate.go:3631
github.com/matrixorigin/matrixone/pkg/frontend.authenticateUserCanExecuteStatementWithObjectTypeAccountAndDatabase
	/home/runner/work/matrixone/matrixone/head/pkg/frontend/authenticate.go:3999
github.com/matrixorigin/matrixone/pkg/frontend.authenticateUserCanExecuteStatement
	/home/runner/work/matrixone/matrixone/head/pkg/frontend/mysql_cmd_executor.go:2312
github.com/matrixorigin/matrixone/pkg/frontend.(*MysqlCmdExecutor).doComQuery
	/home/runner/work/matrixone/matrixone/head/pkg/frontend/mysql_cmd_executor.go:2461
github.com/matrixorigin/matrixone/pkg/frontend.(*MysqlCmdExecutor).ExecRequest
	/home/runner/work/matrixone/matrixone/head/pkg/frontend/mysql_cmd_executor.go:3103
github.com/matrixorigin/matrixone/pkg/frontend.(*Routine).Loop
	/home/runner/work/matrixone/matrixone/head/pkg/frontend/routine.go:159
2022/11/08 13:48:32.338201 +0000 ERROR frontend/util.go:505 query trace status {"connection_id": 4294967295, "statement": "select rp.privilege_id, rp.with_grant_option from mo_catalog.mo_database as d cross join mo_catalog.mo_role_privs as rp where d.dat_id = rp.obj_id and rp.obj_type = \"database\" and rp.role_id = 0 and rp.privilege_id = 21 and rp.privilege_level = \"d\" and d.datname = \"\"", "status": "fail", "error": "rpc timeout", "session_info": ""}
github.com/matrixorigin/matrixone/pkg/frontend.logError
	/home/runner/work/matrixone/matrixone/head/pkg/frontend/util.go:505
github.com/matrixorigin/matrixone/pkg/frontend.logStatementStringStatus
	/home/runner/work/matrixone/matrixone/head/pkg/frontend/util.go:489
github.com/matrixorigin/matrixone/pkg/frontend.logStatementStatus
	/home/runner/work/matrixone/matrixone/head/pkg/frontend/util.go:481
github.com/matrixorigin/matrixone/pkg/frontend.(*MysqlCmdExecutor).doComQuery
	/home/runner/work/matrixone/matrixone/head/pkg/frontend/mysql_cmd_executor.go:3034
github.com/matrixorigin/matrixone/pkg/frontend.(*BackgroundHandler).Exec
	/home/runner/work/matrixone/matrixone/head/pkg/frontend/session.go:1863
github.com/matrixorigin/matrixone/pkg/frontend.determineRoleSetHasPrivilegeSet.func1
	/home/runner/work/matrixone/matrixone/head/pkg/frontend/authenticate.go:3469
github.com/matrixorigin/matrixone/pkg/frontend.determineRoleSetHasPrivilegeSet
	/home/runner/work/matrixone/matrixone/head/pkg/frontend/authenticate.go:3507
github.com/matrixorigin/matrixone/pkg/frontend.determineUserHasPrivilegeSet
	/home/runner/work/matrixone/matrixone/head/pkg/frontend/authenticate.go:3631
github.com/matrixorigin/matrixone/pkg/frontend.authenticateUserCanExecuteStatementWithObjectTypeAccountAndDatabase
	/home/runner/work/matrixone/matrixone/head/pkg/frontend/authenticate.go:3999
github.com/matrixorigin/matrixone/pkg/frontend.authenticateUserCanExecuteStatement
	/home/runner/work/matrixone/matrixone/head/pkg/frontend/mysql_cmd_executor.go:2312
github.com/matrixorigin/matrixone/pkg/frontend.(*MysqlCmdExecutor).doComQuery
	/home/runner/work/matrixone/matrixone/head/pkg/frontend/mysql_cmd_executor.go:2461
github.com/matrixorigin/matrixone/pkg/frontend.(*MysqlCmdExecutor).ExecRequest
	/home/runner/work/matrixone/matrixone/head/pkg/frontend/mysql_cmd_executor.go:3103
github.com/matrixorigin/matrixone/pkg/frontend.(*Routine).Loop
	/home/runner/work/matrixone/matrixone/head/pkg/frontend/routine.go:159
2022/11/08 13:48:32.338318 +0000 INFO frontend/util.go:495 query trace status {"connection_id": 4294967295, "statement": "rollback", "status": "success", "session_info": ""}
2022/11/08 13:48:32.338349 +0000 ERROR frontend/util.go:520 routine send response failed &{1 0 3 0xc06900ae40 0 0 0}. error:illegal state  connectionId 1117
github.com/matrixorigin/matrixone/pkg/frontend.logErrorf
	/home/runner/work/matrixone/matrixone/head/pkg/frontend/util.go:520
github.com/matrixorigin/matrixone/pkg/frontend.(*Routine).Loop
	/home/runner/work/matrixone/matrixone/head/pkg/frontend/routine.go:165
2022/11/08 13:48:32.349819 +0000 INFO disttae/util.go:191 error: internal error: zonemap is nil {"span": {}}
```

**Comment by yingfeng** [comment]
might relate to #6330  @nnsgmsone

**Comment by fengttt** [comment]
dup #6423.

**Comment by sukki37** [comment]
> dup #6423.

error, failed cases, and behavior are totally different, u sure this is duplicated with 6423?

**Comment by daviszhen** [comment]
This error below may be the key. When this error happens, the behaviour of the connection becomes abnormal. 
Further more,after this error happens  every new connection from the client also works abnormally and pitches the some error below.

```
2022/11/08 13:01:59.398295 +0000 INFO disttae/engine.go:97 error: the transaction has been committed or aborted {"span": {}}
```
<img width="1330" alt="6421" src="https://user-images.githubusercontent.com/60595215/200798149-4af343eb-210f-4a8e-bb4c-66fa2a0a28e9.png">

**Comment by yingfeng** [comment]
Update:

#6450 and #6412 will definitely trigger failure

**Comment by yingfeng** [comment]
Update:

#6412 multi cn e2e could succeed after #6464 and #6461 have been merged. @zhangxu19830126 @nnsgmsone @m-schen

**Comment by yingfeng** [comment]
Update

after #6511 #6499 are merged, #6450 could pass the multi cn e2e
